x-pack/plugins/security_solution/public/detection_engine/rule_management/logic/bulk_actions/use_all_rule_ids_for_bulk_action.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { BulkActionType } from '../../../../../common/detection_engine/rule_management/api/rules/bulk_actions/request_schema';
+import { useRulesTableContextOptional } from '../../../rule_management_ui/components/rules_table/rules_table/rules_table_context';
+
+export function useAllRuleIdsForBulkAction(): (bulkActionType: BulkActionType) => string[] {
+  const rulesTableContext = useRulesTableContextOptional();
+
+  return useCallback(
+    (bulkActionType: BulkActionType) => {
+      const allRules = rulesTableContext?.state.isAllSelected ? rulesTableContext.state.rules : [];
```

Comment:
Not sure I understand this logic. If `isAllSelected` is true, we get all rules from the table's state, but why an empty array otherwise? How it works if the user selects only a couple of rules, for example?

Reply:
`performBulkAction` accepts a `query` or `ids`. It's clear what rules are processing if `ids` is passed but it's impossible to say anything about processing rules in the easy way when a `query` is passed. To handle a specific case when all rules are selected and a query is passed `isAllSelected` is used to detect that.

I've moved `isAllSelected` to one level up to make the intent clearer.